xarray/conventions.py

```diff
@@ -205,7 +205,9 @@ def infer_datetime_units(dates):
     unique_timedeltas = np.unique(np.diff(dates))
     units = _infer_time_units_from_diff(unique_timedeltas)
     reference_date = dates[0] if len(dates) > 0 else '1970-01-01'
-    return '%s since %s' % (units, pd.Timestamp(reference_date))
+    # always format as 'YYYY-mmmm-ddTHH:MM:SS'
```

Comment:
the CF conventions do not specify the use of the `T`, so not sure which convention should take preference (CF vs ISO)